### PR TITLE
close the grpc server in Test_Run_Positive_Register

### PR DESCRIPTION
**Pull request by songlh**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

A grpcserver is started, but not stopped in Test_Run_Positive_Register(), causing some goroutines leaks. The patch is to shut down the grpc server and clean up the blocked goroutines. 

#### Which issue(s) this PR fixes:
None

#### Special notes for your reviewer:

None

#### Does this PR introduce a user-facing change?
None

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

None

### CODE REVIEW

**Comment by cyclinder** [comment]
> Which issue(s) this PR fixes:
Please add a link to the issue fixed by this PR

File: pkg/kubelet/pluginmanager/reconciler/reconciler_test.go [hunk 1]
```diff
@@ -192,6 +192,7 @@ func Test_Run_Positive_Register(t *testing.T) {
 	pluginName := fmt.Sprintf("example-plugin")
 	p := pluginwatcher.NewTestExamplePlugin(pluginName, registerapi.DevicePlugin, socketPath, supportedVersions...)
 	require.NoError(t, p.Serve("v1beta1", "v1beta2"))
+	defer p.Stop()
```
Comment (dims) [inline pkg/kubelet/pluginmanager/reconciler/reconciler_test.go:195]: did you see how this is done in other spots in this file?
https://cs.k8s.io/?q=p.Stop%5C(%5C)&i=nope&files=pkg%2Fkubelet%2Fpluginmanager%2Fpluginwatcher%2Fplugin_watcher_test.go&excludeFiles=&repos=

they use `require.NoError` .. is that something we can do here?

**Comment by SergeyKanzhelev** [comment]
ok, I gave it a try, but it looks like the change is breaking tests. Please re-open if you plan to continue working on this

/close

**Comment by cyclinder** [comment]
@SergeyKanzhelev Hi, Can I work on this?

**Pull request closed without merging**